Mention provider's returned versions in Azure Native Version Guide (#394)
pulumi/pulumi-hugo: themes/default/content/docs/intro/cloud-providers/azure/version-guide.md

Each top-level resource maps internally to a specific API version. The native Azure provider maintains this mapping with the following guidelines:

- When a new major version of the provider is released, the API versions are set to the latest stable API version available at that moment. If no stable version is published yet, the newest preview version is used.

> - Azure API reports a list of supported API versions per resource and resource provider. We hold off promoting resources to new API versions until they are reported as supported as seen via the `az provider list` command.

Reviewer: ```suggestion
- Azure API reports a list of supported API versions per resource and resource provider. We promote resources to new API versions once they show up as supported with the `az provider list` command.
```
